manager/apis/app/v1alpha1/blueprint_types.go

```diff
@@ -5,87 +5,60 @@ package v1alpha1
 
 import (
 	"fybrik.io/fybrik/pkg/model/taxonomy"
+	"github.com/rs/zerolog"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
-// CopyModuleArgs define the input parameters for modules that copy data from location A to location B
-// Credentials are stored in a credential management system such as vault
-type CopyModuleArgs struct {
-
+// AssetContext defines the input parameters for modules that access an asset
+type AssetContext struct {
 	// Source is the where the data currently resides
-	// +required
-	Source DataStore `json:"source"`
-
-	// Destination is the data store to which the data will be copied
-	// +required
-	Destination DataStore `json:"destination"`
-
-	// AssetID identifies the asset to be used for accessing the data when it is ready
-	// It is copied from the FybrikApplication resource
-	// +required
-	AssetID string `json:"assetID"`
-
-	// Transformations are different types of processing that may be done to the data as it is copied.
 	// +optional
-	Transformations []taxonomy.Action `json:"transformations,omitempty"`
-}
+	Source *DataStore `json:"source,omitempty"`
 
-// ReadModuleArgs define the input parameters for modules that read data from location A
-type ReadModuleArgs struct {
-	// Source of the read path module
-	// +required
-	Source DataStore `json:"source"`
+	// Destination is the data store to which the data will be written
+	// +optional
+	Destination *DataStore `json:"destination,omitempty"`
 
 	// AssetID identifies the asset to be used for accessing the data when it is ready
 	// It is copied from the FybrikApplication resource
 	// +required
 	AssetID string `json:"assetID"`
 
-	// Transformations are different types of processing that may be done to the data
+	// Transformations are different types of processing that may be done to the data as it is copied.
 	// +optional
 	Transformations []taxonomy.Action `json:"transformations,omitempty"`
 }
 
-// WriteModuleArgs define the input parameters for modules that write data to location B
-type WriteModuleArgs struct {
-	// Destination is the data store to which the data will be written
-	// +required
-	Destination DataStore `json:"destination"`
-
-	// AssetID identifies the asset to be used for accessing the data when it is ready
-	// It is copied from the FybrikApplication resource
+type ApplicationDetails struct {
+	// UUID of FybrikApplication
 	// +required
-	AssetID string `json:"assetID"`
-
-	// Transformations are different types of processing that may be done to the data as it is written.
-	// +optional
-	Transformations []taxonomy.Action `json:"transformations,omitempty"`
-}
+	UUID string `json:"uuid"`
```

Comment:
It's in the label isn't it?  Why do we need it here as well?

Reply:
I'll remove it.

Reply:
Removed uuid and labels from blueprint_types. UUID is obtained from blueprint annotations, labels are in the blueprint resource itself.